experimental/packages/api-logs/src/types/Logger.ts

```diff
@@ -14,21 +14,56 @@
  * limitations under the License.
  */
 
-import { LogRecord } from './LogRecord';
-import { LogEvent } from './LogEvent';
+import type { Attributes, SpanContext, TimeInput } from "@opentelemetry/api";
+
+import type { LogEvent, LogRecord, SeverityNumber } from "./LogRecord";
+
+export interface LogRecordOptions {
+  /**
+   * The time when the log record occurred as UNIX Epoch time in nanoseconds.
+   */
+  timestamp?: TimeInput;
+
+  /**
+   * Numerical value of the severity.
+   */
+  severityNumber?: SeverityNumber;
+
+  /**
+   * The severity text.
+   */
+  severityText?: string;
+
+  /**
+   * A value containing the body of the log record.
+   */
+  body?: string;
+
+  /**
+   * Attributes that define the log record.
+   */
+  attributes?: Attributes;
+
+  /**
+   * The trace context.
+   */
+  context?: SpanContext;
+}
 
 export interface Logger {
   /**
-   * Emit a log record. This method should only be used by log appenders.
+   * Get a log record. This method should only be used by log appenders.
    *
-   * @param logRecord
+   * @param options
    */
-  emitLogRecord(logRecord: LogRecord): void;
+  getLogRecord(options?: LogRecordOptions): LogRecord;
```

Comment:
Why make this change? I don't see this specified anywhere

Reply:
Yes, there are a few main issues to consider here
1. Consistency of style (major); Logger is used to create data in opentelemetry, such as: Meter.createCounter or Tracer.startSpan, both of which return a Span /Couter object, but the data is committed through span.end/couter.add. So Logger.getLogRecrd and LogRecord.emit are also assigned similar responsibilities
2. User experience of developers; emit directly assembled data, can not provide developers with flexible Settings and modification capabilities, such as setAtributes and so on
3. Of course, there is also a small reason to keep the same style with other languages. Here, I also refer to the implementation of opentelemtry-java (which has implemented the log api), and they also handle it in this way
<img width="919" alt="image" src="https://user-images.githubusercontent.com/12229677/204718686-f7658ac9-70b6-4912-91c1-2e641cc5fc83.png">

Reply:
I have originally proposed a similar shape of the API with createLogRecord() and createEvent() methods. But we decided to opt for a simpler API instead. This was discussed [here](https://github.com/open-telemetry/opentelemetry-js/pull/3117#pullrequestreview-1061848788).

I don't think we need to be consistent with the Tracer and Meter APIs, as those need to have two different calls while emitting logs does not. The only advantage would be to have methods like `setAttribute()` to build an instance of a LogRecord over time. But that's just a convenience, IMO. The counterargument is to have a simpler API, as most logging libraries do.

Reply:
Ok, thank you for your reply. I also considered it when I handled it before. I was struggling to decide whether to use a simpler api or keep the same trace/metric style. That is, consistency is a better priority, and now that we've discussed it here we'll use the solution that we've discussed, and I'm going to roll back that part of the code.